packages/tscc/src/spawn_compiler.ts

```diff
@@ -13,24 +18,45 @@ export default function spawnCompiler(providedArgs: string[], logger: Logger, de
 	// TODO consider moving this to tscc.ts.
 	compilerProcess.stderr.on('data', (data) => {
 		logger.log(data);
-	})
+	});
+	
 	compilerProcess.on('error', (err) => {
 		logger.log(chalk.red(`Closure compiler spawn error, Is java in your path?\n${err.message}`));
 		//	onClose(1);
 	});
+
 	return compilerProcess;
 }
 
-function getSupportedCompiler() {
```

Comment:
Why have you replaced a simple enum access to switch statement? Also, it changes a behavior of a program -- It now throws when platform is not `darwin`, `win32`, or `linux`. Previously it just fallbacks to running java version.

Reply:
I patched 80 type mismatches in about 3 hours, I was just working on the fly. The compiler complained about the existing code, so it was modified until it did not complain. I will update this.